clients/src/main/java/org/apache/kafka/clients/consumer/internals/Coordinator.java

```diff
@@ -49,38 +43,30 @@
 
 import java.io.Closeable;
 import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
-import java.util.concurrent.TimeUnit;
 
 /**
  * This class manages the coordination process with the consumer coordinator.
  */
-public final class Coordinator implements Closeable {
+public final class Coordinator extends GroupCoordinator<ConsumerProtocol.Subscription, ConsumerProtocol.Assignment>
+        implements Closeable, Metadata.Listener {
 
     private static final Logger log = LoggerFactory.getLogger(Coordinator.class);
 
-    private final ConsumerNetworkClient client;
-    private final Time time;
-    private final String groupId;
-    private final Heartbeat heartbeat;
-    private final HeartbeatTask heartbeatTask;
-    private final int sessionTimeoutMs;
-    private final String assignmentStrategy;
+    private final Map<String, PartitionAssignor> assignors;
+    private final org.apache.kafka.clients.Metadata metadata;
+    private final MetadataSnapshot metadataSnapshot;
```

Comment:
This metadataSnapshot is only written upon metadata update, but not read in doSync: we still use the above metadata object, and actually it seems not used anywhere. Is this intentional?